### PR TITLE
Cache payload in different directory to v1

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

Caches session payloads in a different directory to V1. I've taken the approach of reusing the logic of the `PayloadStorageService` with a different instance. The `IntakeService` receives a request to cache a payload & then submits a job. IMO ultimately this will be the simplest way to appropriately prioritise writing payloads/caches, and to avoid repeating serialization/gzipping code in multiple places throughout the app.

## Testing

Added unit tests.

### CODE REVIEW

File: embrace-android-delivery/src/main/kotlin/io/embrace/android/embracesdk/internal/delivery/intake/IntakeService.kt [hunk 1]
```diff
@@ -20,4 +20,9 @@ interface IntakeService : Shutdownable {
      * Stores a payload on disk as its JSON representation.
      */
     fun take(intake: Envelope<*>, metadata: StoredTelemetryMetadata)
+
+    /**
+     * Caches a payload on disk as its JSON representation.
+     */
+    fun cache(intake: Envelope<*>, metadata: StoredTelemetryMetadata)
```
Comment (bidetofevil) [inline embrace-android-delivery/src/main/kotlin/io/embrace/android/embracesdk/internal/delivery/intake/IntakeService.kt:27]: Do we need another method in the interface if we can tell whether a payload is complete to determine whether it's a cache or take scenario?

**Review by bidetofevil — APPROVED** [review]
This is fine to me if it's clearer to you at the interface level that `take` and `cache` be two separate methods. Initially I thought I preferred just one method but reading the usage, this makes the intentions more clear, so now I don't know what I prefer 😅

File: embrace-android-delivery/src/main/kotlin/io/embrace/android/embracesdk/internal/delivery/intake/IntakeServiceImpl.kt [hunk 2]
```diff
@@ -23,16 +24,28 @@ class IntakeServiceImpl(
 
     override fun take(intake: Envelope<*>, metadata: StoredTelemetryMetadata) {
         worker.submit(metadata) {
-            processIntake(intake, metadata)
+            processIntake(intake, metadata, payloadStorageService)
```
[the remark this answers is not in the document]
Comment (fractalwrench) [reply embrace-android-delivery/src/main/kotlin/io/embrace/android/embracesdk/internal/delivery/intake/IntakeServiceImpl.kt:27]: Yes. Persisting snapshots should be lower priority than capturing other data IMO.

File: embrace-android-delivery/src/main/kotlin/io/embrace/android/embracesdk/internal/delivery/intake/IntakeService.kt (hunk 1 above)
Comment (fractalwrench) [reply embrace-android-delivery/src/main/kotlin/io/embrace/android/embracesdk/internal/delivery/intake/IntakeService.kt]: Updated to just use one method.

**Pull request merged**